extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -1349,10 +1353,10 @@ private void handleParseException(ParseException pe, ConsumerRecord<byte[], byte
   private Map<String, Object> getTaskCompletionRowStats()
   {
     Map<String, Object> metrics = Maps.newHashMap();
-    if (metricsGetter != null) {
+    if (rowIngestionMeters != null) {
```

Comment:
the fact that there are some null checks but not others, and I'm not entirely clear if the `rowIngestionMeters` can be accessed in multiple threads, leads me to suggest that the `rowIngestionMeters` be created as `final` in the constructor and accessed everywhere. Is the overhead small enough to where that shouldn't be a problem?

Reply:
moved these to constructors